### PR TITLE
Small changes in Actions

### PR DESCRIPTION
**Pull request by GiudGiud**
Aside from the boundary integration / value functors which are using for connecting components and arent needed right away, this might be all I can extract from the Physics PR without actually introducing Physics

### CODE REVIEW

File: framework/include/actions/ActionWarehouse.h [hunk 1]
```diff
@@ -120,7 +120,18 @@ class ActionWarehouse : public ConsoleStreamInterface
...
     if (!p)
-      mooseError("Action with name being ", name, " does not exist");
+    {
+      std::vector<std::string> all_names;
+      for (auto act_ptr : _all_ptrs)
+        if (!act_ptr->name().empty())
+          all_names.push_back(act_ptr->name());
+        else
+          all_names.push_back("unnamed");
+      mooseError("Action with name being ",
```
Comment (lindsayad) [inline framework/include/actions/ActionWarehouse.h:130]:
```suggestion
      mooseError("Action with name '",
```

File: framework/include/actions/ActionWarehouse.h [hunk 2]
```diff
@@ -120,7 +120,18 @@ class ActionWarehouse : public ConsoleStreamInterface
...
+                 name,
+                 " does not exist.\n Actions that do exist:",
```
Comment (lindsayad) [inline framework/include/actions/ActionWarehouse.h:132]:
```suggestion
                 "' does not exist.\n These are the Actions that do exist:",
```

File: framework/include/actions/Action.h [hunk 3]
```diff
@@ -161,6 +161,11 @@ class Action : public ConsoleStreamInterface,
 
   inline bool isParamValid(const std::string & name) const { return _pars.isParamValid(name); }
 
+  inline bool isParamSetByUser(const std::string & name) const
+  {
+    return _pars.isParamSetByUser(name);
+  }
```
Comment (lindsayad) [inline framework/include/actions/Action.h:167]: Inline in the header but outside of the class definition would be my preference but as there are many other methods like this currently, it's not a requirement

File: framework/include/actions/Action.h [hunk 4]
```diff
@@ -286,3 +288,9 @@ Action::getParam(const std::string & name) const
 {
   return InputParameters::getParamHelper(name, _pars, static_cast<T *>(0));
 }
+
+bool
```
Comment (lindsayad) [inline framework/include/actions/Action.h:292]:
```suggestion
inline bool
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]
Learend something new today. In-line declaration is enough

**Comment by GiudGiud** [comment]
but we cant inline code that is declared in the header and defined in the source right

**Comment by lindsayad** [comment]
Well if you put the definition in the C file then you will probably get undefined symbols or references because the compiler won’t generate code that can be used/referred-to by other translation units

**Pull request merged**